src/commands/tokens/delete.ts

```diff
@@ -27,22 +33,42 @@ export default class AppDeleteCommand extends Command {
   @position(0)
   id: string;
 
+  @help("The type of token principal authentication: [" + allPrincipalTypes.join(", ") + "]")
+  @shortName("t")
+  @longName("type")
+  @hasArg
+  @defaultValue("user")
+  public principalType: string;
+
   async run(client: AppCenterClient): Promise<CommandResult> {
-    const confirmation = await prompt.confirm(`Do you really want to delete the token with ID "${this.id}"`);
+    const tokenLevel = this.principalType === PrincipalType.USER ? principalMessaging.user : principalMessaging.app;
```

Comment:
Can we have a validation function that verifies that an acceptable option is passed? We want to have exactly `user` or `app`, or unspecified (default).
Some commands have `validateParameters()`.